### DISCUSSION TITLE
Dynamic content blocks - what to use?

### DISCUSSION THREAD

**Discussion by katoen** [atmostfear-entertainment/aeon#24] [Q&A] [answered]
Hi, I'm testing Aegis and I must say: it looks really promissing! Thank you for this tremendous effort  @alexdeborba @atmostfear-entertainment !

I'm currently checking it out for a future rebuild of an event venue website. It's using FooEvents, so it's WooCommerce based. 
For now I feel a lot of my build is possible without installing too many additional plugins. However I do need to be able to add some **dynamic content** (like an event date which is stored as a (fooevents) custom field in the product). 

I have several advanced block plugins and themes in my stack (Greenshift, Generateblocks, Cwicly,...) but I would rather not bloat my FSE experiences with a pletoria of unneeded blocks.

Can anyone recommend a **lean plugin to just add some dynamic content**? Or maybe Aegis integrates with a preferred block plugin?

🙏

**Comment by katoen** [comment]
I prefer not to use the shortcode block, or would this be the most logic option?

**Comment by alexdeborba** [comment, marked as answer]
Hello @katoen,

Thank you for your kind words. They are truly encouraging for our team.

Pertaining to FooEvents, we are in the process of redesigning one of our company websites, which will showcase both in-person and virtual events, complete with ticket sales. We have chosen to use FooEvents, primarily because we are organizing our own events. This aligns perfectly with your requirements, and as Aegis will often evolve based on our own implementation and visual needs, we will be tailoring it to support FooEvents seamlessly.

I recognize the importance of Dynamic Content. While we do have plans to integrate dynamic content, we aim to do so without relying on shortcodes, and by adopting a unique approach that still maintains the foundational blocks offered by WordPress.

Instead of using a companion blocks plugin, we will be focusing on the core blocks. In addition to the features you mentioned, we also have intentions to introduce Full Pages dedicated to event displays, adaptable checkout processes for ticket purchases, dedicated pages for speakers, and more.

I am aligned with the notion of avoiding the addition of block plugins directly to the theme. Rest assured, Aegis will always incorporate features directly within the theme, negating the need for separate block installations. This aligns with the comprehensive and streamlined vision we have set for Aegis.

Although we have not fully delved deeply into FooEvents yet, I would recommend submitting a [Feature Request](https://github.com/atmostfear-entertainment/aegis/issues/new/choose). This will enable us to understand your specific requirements and potentially address them in a future release. Doing so not only helps us offer unique solutions but also upholds our commitment to being community-driven.

**Reply by katoen** [reply]
> Pertaining to FooEvents, we are in the process of redesigning one of our company websites, which will showcase both in-person and virtual events, complete with ticket sales. 

This is excellent news. Would love to have a modern block theme that has a high level of Fooevent compatibility 👍 
I've been using Fooevents for over 4 years, solely for real-life events. It's proven to be an excellent and reliable choice. We've successfully sold over 30,000 tickets with minimal problems. 
However, my only gripe is that the system lacks built-in extensibility and doesn't offer options for integrating with automation platforms. I'd really wish they would adopt an API approach, allowing customization to better suit my needs (such as integrating with Flowmattic).

I'm uncertain about the implementation of the dynamic content feature you describe. It has always seemed to me that enhancing core blocks to make them dynamic is the most efficient approach. However, it appears that this isn't an adopted practice in other block plugins, leading me to believe it might not be possible. It would be fantastic if this somehow could be addressed and resolved in your theme.

At the moment, it seems you don't have a recommended third-party block plugin that enable the integration of dynamic data into the site. As an interim solution, I plan to utilize shortcodes. Once I've developed a proof of concept for my site, I'll outline my requirements in a feature request.

**Reply by katoen** [reply]
> As an interim solution, I plan to utilize shortcodes. 

Update: shortcode block doesn't seem to work properly inside the loop, but I found this plugin: https://wordpress.org/plugins/display-a-meta-field-as-block/
